### PR TITLE
move and delete account

### PR DESCRIPTION
**Pull request by ArtificialOwl**

### CODE REVIEW

File: lib/Command/AccountDelete.php [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+	) {
+		parent::__construct();
+
+		$this->userManager = $userManager;
+		$this->accountService = $accountService;
+		$this->cacheActorService = $cacheActorService;
+		$this->configService = $configService;
+	}
+
+	protected function configure() {
```
Comment (CarlSchwan) [inline lib/Command/AccountDelete.php:66]:
```suggestion
	protected function configure(): void {
```

File: lib/Db/ActionsRequest.php [hunk 2]
```diff
@@ -153,13 +153,21 @@ public function delete(ACore $item) {
...
-//		$qb = $this->getActionsDeleteSql();
-//		$this->limitToObjectId($qb, $objectId);
-//
-//		$qb->execute();
-//	}
+	public function deleteByActor(string $actorId): void {
+		$qb = $this->getActionsDeleteSql();
+		$qb->limitToDBField('actor_id_prim', $qb->prim($actorId));
+
+		$qb->execute();
```
Comment (CarlSchwan) [inline lib/Db/ActionsRequest.php:160]:
```suggestion
		$qb->executeStatement();
```

File: lib/Db/ActionsRequest.php [hunk 3]
```diff
@@ -153,13 +153,21 @@ public function delete(ACore $item) {
...
+
+
+	public function moveAccount(string $actorId, string $newId): void {
+		$qb = $this->getActionsUpdateSql();
+		$qb->set('actor_id', $qb->createNamedParameter($newId))
+		   ->set('actor_id_prim', $qb->createNamedParameter($qb->prim($newId)));
+
+		$qb->limitToDBField('actor_id_prim', $qb->prim($actorId));
+
+		$qb->execute();
```
Comment (CarlSchwan) [inline lib/Db/ActionsRequest.php:171]:
```suggestion
		$qb->executeStatement();
```

File: lib/Db/CacheDocumentsRequest.php [hunk 4]
```diff
@@ -239,4 +241,22 @@ public function deleteById(string $id) {
 
 		$qb->execute();
 	}
+
+	public function deleteByParent(string $parentId) {
```
Comment (CarlSchwan) [inline lib/Db/CacheDocumentsRequest.php:245]:
```suggestion
	public function deleteByParent(string $parentId): void {
```

File: lib/Db/CacheDocumentsRequest.php [hunk 5]
```diff
@@ -239,4 +241,22 @@ public function deleteById(string $id) {
...
+
+
+	public function moveAccount(string $actorId, string $newId): void {
+		$qb = $this->getCacheDocumentsUpdateSql();
+		$qb->set('parent_id', $qb->createNamedParameter($newId))
+		   ->set('parent_id_prim', $qb->createNamedParameter($qb->prim($newId)));
+
+		$qb->limitToDBField('parent_id_prim', $qb->prim($actorId));
+
+		$qb->execute();
```
Comment (CarlSchwan) [inline lib/Db/CacheDocumentsRequest.php:260]:
```suggestion
		$qb->executeStatement();
```

File: lib/Db/FollowsRequest.php [hunk 6]
```diff
@@ -304,4 +301,36 @@ public function deleteById(string $id) {
 
 		$qb->execute();
 	}
+
+
+	/**
+	 * @param string $actorId
+	 * @param Person $new
+	 */
+	public function moveAccountFollowers(string $actorId, Person $new) {
```
Comment (CarlSchwan) [inline lib/Db/FollowsRequest.php:310]:
```suggestion
	public function moveAccountFollowers(string $actorId, Person $new): void {
```

File: lib/Db/FollowsRequest.php [hunk 7]
```diff
@@ -304,4 +301,36 @@ public function deleteById(string $id) {
...
+
+		$qb->executeStatement();
+	}
+
+
+	/**
+	 * @param string $actorId
+	 * @param Person $new
+	 */
+	public function moveAccountFollowing(string $actorId, Person $new) {
```
Comment (CarlSchwan) [inline lib/Db/FollowsRequest.php:327]:
```suggestion
	public function moveAccountFollowing(string $actorId, Person $new): void {
```

File: lib/Db/RequestQueueRequest.php [hunk 8]
```diff
@@ -197,4 +199,21 @@ public function delete(RequestQueue $queue): void {
 
 		$qb->executeStatement();
 	}
+
+	public function deleteByAuthor(string $actorId) {
```
Comment (CarlSchwan) [inline lib/Db/RequestQueueRequest.php:203]:
```suggestion
	public function deleteByAuthor(string $actorId): void {
```

File: lib/Db/StreamRequest.php [hunk 9]
```diff
@@ -772,6 +789,19 @@ public function deleteByAuthor(string $actorId) {
...
+	/**
+	 * @param string $actorId
+	 */
+	public function updateAuthor(string $actorId, string $newId) {
+		$qb = $this->getStreamUpdateSql();
+		$qb->set('attributed_to', $qb->createNamedParameter($newId))
+		   ->set('attributed_to_prim', $qb->createNamedParameter($qb->prim($newId)));
+		$qb->limitToAttributedTo($actorId, true);
+
+		$qb->execute();
```
Comment (CarlSchwan) [inline lib/Db/StreamRequest.php:801]:
```suggestion
		$qb->executeStatement();
```

File: lib/Db/CoreRequestBuilder.php [hunk 10]
```diff
@@ -1263,10 +1264,10 @@ protected function assignDetails(Person $actor, array $data) {
 	 * this just empty all tables from the app.
 	 */
 	public function emptyAll() {
-		$schema = new SchemaWrapper(Server::get(IDBConnection::class));
+		$schema = new SchemaWrapper(Server::get(Connection::class));
```
Comment (artonge) [inline lib/Db/CoreRequestBuilder.php:1267]: @ArtificialOwl Why was it changed ? Looks like psalm is not happy.

Comment (artonge) [reply lib/Db/CoreRequestBuilder.php:1267]: Same on line 1282

**Pull request merged**